gammapy/background/models.py

```diff
@@ -71,3 +80,820 @@ def evaluate(self, x, y):
+
+        Returns
+        -------
+        bin_pos : `~int`
+            index of the det bin containing the specified det (X, Y) pair
+        bin_edges : `~astropy.units.Quantity`
+            det bin edges (x_lo, x_hi, y_lo, y_hi)
+        """
+        # check shape of det: only 1 pair is accepted
+        nvalues = len(det.flatten())
+        if nvalues != 2:
+            raise IndexError("Expected exactly 2 values for det (X, Y), got {}."
+                             .format(nvalues))
+
+        # check that the specified det is within the boundaries of the model
+        det_extent = self.image_extent
+        check_x_lo = (det_extent[0] <= det[0])
+        check_x_hi = (det[0] < det_extent[1])
+        check_y_lo = (det_extent[2] <= det[1])
+        check_y_hi = (det[1] < det_extent[3])
+        if not (check_x_lo and check_x_hi) or not (check_y_lo and check_y_hi):
+            raise ValueError("Specified det {0} is outside the boundaries {1}."
+                             .format(det, det_extent))
+
+        detx_edges_low = self.detx_bins[:-1]
+        detx_edges_high = self.detx_bins[1:]
+        dety_edges_low = self.dety_bins[:-1]
+        dety_edges_high = self.dety_bins[1:]
+	bin_pos_x = np.searchsorted(detx_edges_high, det[0])
+	bin_pos_y = np.searchsorted(dety_edges_high, det[1])
+        bin_pos = np.array([bin_pos_x, bin_pos_y])
+        bin_edges = Angle([detx_edges_low[bin_pos[0]], detx_edges_high[bin_pos[0]],
+                           dety_edges_low[bin_pos[1]], dety_edges_high[bin_pos[1]]])
+
+        return bin_pos, bin_edges
+
+
+    def find_energy_bin(self, energy):
+        """Find the bin that contains the specified energy value.
+
+        TODO: implement test as suggested in:
+            https://github.com/gammapy/gammapy/pull/292#discussion_r33843508
+
+        Parameters
+        ----------
+        energy : `~astropy.units.Quantity`
+            energy to search for
+
+        Returns
+        -------
+        bin_pos : `~int`
+            index of the energy bin containing the specified energy
+        bin_edges : `~astropy.units.Quantity`
+            energy bin edges [E_min, E_max)
+        """
+        # check shape of energy: only 1 value is accepted
+        nvalues = len(energy.flatten())
+        if nvalues != 1:
+            raise IndexError("Expected exactly 1 value for energy, got {}."
+                             .format(nvalues))
+
+        # check that the specified energy is within the boundaries of the model
+        energy_extent = self.spectrum_extent
+        if not (energy_extent[0] <= energy) and (energy < energy_extent[1]):
+            ss_error = "Specified energy {}".format(energy)
+            ss_error += " is outside the boundaries {}.".format(energy_extent)
+            raise ValueError(ss_error)
+
+        energy_edges_low = self.energy_bins[:-1]
+        energy_edges_high = self.energy_bins[1:]
+	bin_pos = np.searchsorted(energy_edges_high, energy)
```

Comment:
Again, looks mis-indented.
Is it? Or do you have tabs?
The `pep8` command line tool will tell you ...